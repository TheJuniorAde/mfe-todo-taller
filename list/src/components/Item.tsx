import React from "react"

import { TodoItem } from "host/Context"

interface ItemProps {
  todo: TodoItem
  updateTodo: (item: TodoItem) => void
}

export const Item: React.FC<ItemProps> = ({ todo, updateTodo }) => {
  const isActive = todo.status === "incomplete"
  const getStatusClassName = (status: TodoItem["status"]) =>
    `rounded-full ${
      status === "completed" ? "bg-emerald-500/50" : "bg-rose-700/50"
    }`

  return (
    <li
      data-testid={`list-item-${todo.id}`}
      key={todo.id}
      className="flex justify-between gap-x-6 px-5 py-5 hover:bg-gray-100 hover:cursor-pointer"
    >
      <label
        data-testid={`list-item-label-${todo.id}`}
        className="flex justify-between w-full hover:cursor-pointer"
      >
        <div className="flex min-w-0 gap-x-4">
          <input
            checked={!isActive}
            onChange={() => updateTodo(todo)}
            className="mr-2 leading-tight"
            type="checkbox"
          />
          <div className="min-w-0 flex-auto">
            <p
              className={`text-sm font-semibold leading-6 text-gray-900 ${
                !isActive && "line-through"
              }`}
            >
              {todo.description}
            </p>
            <p className="mt-1 truncate text-xs leading-5 text-gray-500">
              {isActive
                ? `Created at ${todo.createdAt}`
                : `Finished at ${todo.finishedAt}`}
            </p>
          </div>
        </div>
        <div className="hidden shrink-0 sm:flex sm:flex-col sm:items-end justify-center">
          <div className="mt-1 flex items-center gap-x-1.5">
            <div className={`flex-none ${getStatusClassName(todo.status)} p-1`}>
              <div
                className={`h-5 w-5 ${getStatusClassName(todo.status)}`}
              ></div>
            </div>
            <p className="text-xs leading-5 text-gray-500">{todo.status}</p>
          </div>
        </div>
      </label>
    </li>
  )
}
