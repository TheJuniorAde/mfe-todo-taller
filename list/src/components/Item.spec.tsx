/**
 * @jest-environment jsdom
 */
import React from "react"
import { render } from "@testing-library/react"
import { Item } from "./Item"

describe("Item", () => {
  it("renders correctly when incomplete", () => {
    const { container } = render(
      <Item
        todo={{
          description: "test",
          id: "",
          createdAt: new Date().toLocaleString("en-US"),
          finishedAt: "",
          status: "incomplete",
        }}
        updateTodo={jest.fn()}
      />
    )

    expect(container.parentElement).toMatchSnapshot()
  })

  it("renders correctly when completed", () => {
    const { container } = render(
      <Item
        todo={{
          description: "test",
          id: "",
          createdAt: "",
          finishedAt: new Date().toLocaleString("en-US"),
          status: "completed",
        }}
        updateTodo={jest.fn()}
      />
    )

    expect(container.parentElement).toMatchSnapshot()
  })
})
