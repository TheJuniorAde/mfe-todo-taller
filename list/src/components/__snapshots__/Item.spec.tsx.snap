// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders correctly when completed 1`] = `
<body>
  <div>
    <li
      class="flex justify-between gap-x-6 px-5 py-5 hover:bg-gray-100 hover:cursor-pointer"
      data-testid="list-item-"
    >
      <label
        class="flex justify-between w-full hover:cursor-pointer"
        data-testid="list-item-label-"
      >
        <div
          class="flex min-w-0 gap-x-4"
        >
          <input
            checked=""
            class="mr-2 leading-tight"
            type="checkbox"
          />
          <div
            class="min-w-0 flex-auto"
          >
            <p
              class="text-sm font-semibold leading-6 text-gray-900 line-through"
            >
              test
            </p>
            <p
              class="mt-1 truncate text-xs leading-5 text-gray-500"
            >
              Finished at 6/4/2024, 6:21:01 PM
            </p>
          </div>
        </div>
        <div
          class="hidden shrink-0 sm:flex sm:flex-col sm:items-end justify-center"
        >
          <div
            class="mt-1 flex items-center gap-x-1.5"
          >
            <div
              class="flex-none rounded-full bg-emerald-500/50 p-1"
            >
              <div
                class="h-5 w-5 rounded-full bg-emerald-500/50"
              />
            </div>
            <p
              class="text-xs leading-5 text-gray-500"
            >
              completed
            </p>
          </div>
        </div>
      </label>
    </li>
  </div>
</body>
`;

exports[`Item renders correctly when incomplete 1`] = `
<body>
  <div>
    <li
      class="flex justify-between gap-x-6 px-5 py-5 hover:bg-gray-100 hover:cursor-pointer"
      data-testid="list-item-"
    >
      <label
        class="flex justify-between w-full hover:cursor-pointer"
        data-testid="list-item-label-"
      >
        <div
          class="flex min-w-0 gap-x-4"
        >
          <input
            class="mr-2 leading-tight"
            type="checkbox"
          />
          <div
            class="min-w-0 flex-auto"
          >
            <p
              class="text-sm font-semibold leading-6 text-gray-900 false"
            >
              test
            </p>
            <p
              class="mt-1 truncate text-xs leading-5 text-gray-500"
            >
              Created at 6/4/2024, 6:21:01 PM
            </p>
          </div>
        </div>
        <div
          class="hidden shrink-0 sm:flex sm:flex-col sm:items-end justify-center"
        >
          <div
            class="mt-1 flex items-center gap-x-1.5"
          >
            <div
              class="flex-none rounded-full bg-rose-700/50 p-1"
            >
              <div
                class="h-5 w-5 rounded-full bg-rose-700/50"
              />
            </div>
            <p
              class="text-xs leading-5 text-gray-500"
            >
              incomplete
            </p>
          </div>
        </div>
      </label>
    </li>
  </div>
</body>
`;
